pkg/pstore/api.go

```diff
@@ -0,0 +1,149 @@
+// Copyright (C) 2018 Storj Labs, Inc.
+// See LICENSE for copying information.
+
+package pstore // import "storj.io/storj/pkg/pstore"
+
+import (
+	"bufio"
+	"fmt"
+	"io"
+	"os"
+	"path"
+)
+
+type argError struct {
+	arg string
+	msg string
+}
+
+func (e *argError) Error() string {
+	return fmt.Sprintf("HashError (%s): %s", string(e.arg), e.msg)
+}
+
+type fsError struct {
+	path string
+	msg  string
+}
+
+func (e *fsError) Error() string {
+	return fmt.Sprintf("FSError (%s): %s", e.path, e.msg)
+}
+
+func Store(hash string, r *bufio.Reader, dir string) error {
+	fmt.Println("Storing...")
+	if len(hash) < 20 {
+		return &argError{hash, "Hash is too short"}
+	}
+	if dir == "" {
+		return &argError{dir, "No path provided"}
+	}
+
+	// Folder structure
+	folder1 := string(hash[0:2])
+	folder2 := string(hash[2:4])
+	fileName := string(hash[4:])
+
+	// Create directory path string
+	dirpath := path.Join(dir, folder1, folder2)
+
+	// Create directory path on file system
+	mkDirerr := os.MkdirAll(dirpath, 0700)
```

Comment:
github comment lost due to gofmt (good job!) but yeah, convention is to use `err` as the variable name for all errors